kolibri/plugins/learn/assets/src/modules/examViewer/handlers.js

```diff
@@ -32,14 +33,17 @@ export function showExam(store, params, alreadyOnQuiz) {
     store.commit('CORE_SET_PAGE_LOADING', false);
   } else {
     const promises = [
```

Comment:
could you try using the `ClassroomResource`? You can import it from `kolibri.resources`. If you only want the name of the class, this is a lighter-weight Viewset, since it doesn't retrieve the lessons and quizzes too.

Another optimization would be to check to see if the `currentClassroom` is defined before applying the `SET_CURRENT_CLASSROOM` mutation.

Reply:
Just confirmed that `ClassroomResource.fetchModel({ id: classId })` works fine at line 6.

Reply:
Yeah, rather than requiring another request - wouldn't this data be available from the class_summary_api call as well?

Reply:
I don't think so, since this is the Learner Quiz viewer, not the Coach quiz viewer (where Class Summary is available)

Reply:
Oh sorry - completely missed that. Disregard!

Reply:
Thanks for the advice everyone, and sorry for the delay. I've changed the handlers to use `ClassroomResource` instead of `LearnerClassroomResource`, and it seems to work great.

As for this suggestion:
> Another optimization would be to check to see if the currentClassroom is defined before making the API call to ClassroomResource, potentially saving a network call.

I definitely agree that this would save an API call, but I'm not sure if the way I'd do it is clean enough. 

Given that the API calls are effectively wrapped into `promises` and called altogether via `ConditionalPromise.all(promises).only`, would it be better readability-wise to have a separate check for `currentClassroom` >> Make a separate call to `ClassroomResource` >> Make the aforementioned `ConditionalPromise` call for the rest of the APIs?

Reply:
Thanks for the advice everyone, and sorry for the delay! I've changed `LearnerClassroomResource` to `ClassroomResource` and it seems to work great. 

As for checking for the existence of `classroomResource` before making a call to the API, I'm not sure if the way I'd implement it would be clean enough. 

It looks like the API calls are wrapped into `promises` and then  effectively 'batch called' with `ConditionalPromise.all(promises).only()`. I think something like this would work: Check if `classroomResource` exists >> Make call to `ClassroomResource` if it doesn't >> Make the rest of the 'batch' calls - but I'm not entirely sure if the extra handful of lines I'd add would be worth the single API call - any advice one way or the other would be great!